internal/daysteps/daysteps.go

```diff
@@ -1,13 +1,57 @@
 package daysteps
 
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Yandex-Practicum/tracker/internal/personaldata"
+	"github.com/Yandex-Practicum/tracker/internal/spentenergy"
+)
+
+// DaySteps содержит данные о дневной прогулке.
 type DaySteps struct {
-	// TODO: добавить поля
+	Steps    int
+	Duration time.Duration
+	personaldata.Personal
 }
 
+// Parse парсит строку с данными о прогулке в формате "шаги,длительность".
 func (ds *DaySteps) Parse(datastring string) (err error) {
-	// TODO: реализовать функцию
+	parts := strings.Split(datastring, ",")
+	if len(parts) != 2 {
+		return fmt.Errorf("неверный формат строки данных")
+	}
+
+	ds.Steps, err = strconv.Atoi(parts[0])
+	if err != nil {
+		return fmt.Errorf("ошибка при парсинге шагов: %w", err)
+	}
+	if ds.Steps <= 0 {
+		return fmt.Errorf("количество шагов должно быть больше нуля")
+	}
+
+	durationStr := parts[1]
+	ds.Duration, err = time.ParseDuration(strings.ReplaceAll(durationStr, "h", "h"))
+	if err != nil {
+		return fmt.Errorf("ошибка при парсинге продолжительности: %w", err)
+	}
+	if ds.Duration <= 0 {
+		return fmt.Errorf("продолжительность хотьбы должна быть больше нуля")
+	}
+	return nil
 }
 
+// ActionInfo формирует строку с информацией о прогулке.
 func (ds DaySteps) ActionInfo() (string, error) {
-	// TODO: реализовать функцию
+	distance := spentenergy.Distance(ds.Steps, ds.Personal.Height)
+
+	calories, err := spentenergy.WalkingSpentCalories(ds.Steps, ds.Weight, ds.Height, ds.Duration)
+	if err != nil {
+		return "", err
```

Comment:
Старайся оборачивать ошибки, чтобы потом проще было понимать откуда они
```suggestion
		return "", return "", fmt.Errorf("error calculating calories: %w", err)
```